account_payment_order/wizard/account_payment_line_create.py

```diff
@@ -0,0 +1,155 @@
+        order = self.env['account.payment.order'].browse(context['active_id'])
+        mode = order.payment_mode_id
+        res.update({
+            'journal_ids': mode.default_journal_ids.ids or False,
+            'target_move': mode.default_target_move,
+            'invoice': mode.default_invoice,
+            'date_type': mode.default_date_type,
+            'payment_mode': mode.default_payment_mode,
+            'order_id': order.id,
+            })
+        return res
+
+    @api.multi
+    def _prepare_move_line_domain(self):
+        self.ensure_one()
+        journals = self.journal_ids or self.env['account.journal'].search([])
+        domain = [('reconciled', '=', False),
+                  ('company_id', '=', self.order_id.company_id.id),
+                  ('journal_id', 'in', journals.ids)]
+        if self.target_move == 'posted':
+            domain += [('move_id.state', '=', 'posted')]
+        if not self.allow_blocked:
+            domain += [('blocked', '!=', True)]
+        if self.date_type == 'due':
+            domain += [
+                '|',
+                ('date_maturity', '<=', self.due_date),
+                ('date_maturity', '=', False)]
+        elif self.date_type == 'move':
+            domain.append(('date', '<=', self.move_date))
+        if self.invoice:
+            domain.append(('invoice_id', '!=', False))
+        if self.payment_mode:
+            if self.payment_mode == 'same':
+                domain.append(
+                    ('payment_mode_id', '=', self.order_id.payment_mode_id.id))
+            elif self.payment_mode == 'same_or_null':
+                domain += [
+                    '|',
+                    ('payment_mode_id', '=', False),
+                    ('payment_mode_id', '=', self.order_id.payment_mode_id.id)]
+
+        if self.order_id.payment_type == 'outbound':
+            # For payables, propose all unreconciled credit lines,
+            # including partially reconciled ones.
+            # If they are partially reconciled with a supplier refund,
+            # the residual will be added to the payment order.
+            #
+            # For receivables, propose all unreconciled credit lines.
+            # (ie customer refunds): they can be refunded with a payment.
+            # Do not propose partially reconciled credit lines,
+            # as they are deducted from a customer invoice, and
+            # will not be refunded with a payment.
+            domain += [
+                ('credit', '>', 0),
+                #  '|',
+                ('account_id.internal_type', '=', 'payable'),
+                #  '&',
+                #  ('account_id.internal_type', '=', 'receivable'),
+                #  ('reconcile_partial_id', '=', False),  # TODO uncomment
+            ]
+        elif self.order_id.payment_type == 'inbound':
+            domain += [
+                ('debit', '>', 0),
+                ('account_id.internal_type', '=', 'receivable'),
```

Comment:
My concern with payment.orders is that it does not seem to be able to handle refunds. So, in the scenario where you have an angry customer that expects a refund (and is not likely to come again - so little chance to reconcile the refund in the future with a new invoice), you cannot create a payment order, due to this added domain. 

IMHO what should only matter here is that with inbound payment orders you pay debit > 0, for receivables and payables, and with outbound payment orders you pay credit > 0, for receiv & payab.